dataprofiler/tests/profilers/test_column_profile_compilers.py

```diff
@@ -70,6 +71,132 @@ def test_add_profilers(self):
         self.assertEqual(3, merged_compiler._profiles['test'])
         self.assertEqual('compiler1', merged_compiler.name)
 
+    
+    def test_diff_primitive_compilers(self):
+        # Test different data types
+        data1 = pd.Series(['-2', '-1', '1', '2'])
+        data2 = pd.Series(["YO YO YO", "HELLO"])
+        compiler1 = col_pro_compilers.ColumnPrimitiveTypeProfileCompiler(data1)
+        compiler2 = col_pro_compilers.ColumnPrimitiveTypeProfileCompiler(data2)
+
+        expected_diff = {
+            'data_type_representation': {
+                'datetime': 'unchanged',
```

Comment:
can we add a test for the datetime change as well?

Reply:
added!